### ISSUE TITLE
A Leaflet plugin to dry up basic layer addition

### ISSUE THREAD

**Issue by mmnoo** [publiclab/leaflet-environmental-layers#276]
Hey guys, 

I am working on a [Leaflet plugin](https://github.com/mmnoo/leaflet-declarative-layers) to make adding common layer types declarative (ie by configuration or by simply describing the layer in a human-readable way code) rather than redundant imperative code like `leaflet-environmental-layers` seems to suffer from. (Have things changed? I havent really done a deep dive since the spring.)

Anyway, its definitely still early days with the plugin and I am wondering:
- If anyone thinks that approach might be a good fit for drying up `leaflet-environmental-layers` and making contributing layers more accessible to first-timers and new coders.
- If anyone is interested in contributing to the plugin? It will require unit tests written with Jasmine, and TypeScript. 
- [If anyone would like to contribute by being a code reviewer](https://github.com/mmnoo/leaflet-declarative-layers/pulls) (I love code reviews from coders of all levels and dont bite!)
- If anyone would like to see data types or functionality added to the plugin (please discuss here or create new issue in the plugin's repo)
- If anyone thinks the concept is stupid for whatever reason. I still want to know! 

Thanks, 
Melissa

**Comment by jywarren** [comment]
YES this is really amazing. I need to read through more carefully but we
are def. in need of more drying up! Thanks Melissa! I'm rushing on this
project in the next 2 weeks or so, but want to dig in a bit more and
understand where the interface points are here. Thank you!

**Comment by sagarpreet-chadha** [comment]
Interesting!

**Comment by mmnoo** [comment]
Just a friendly `ping`. I added a better readme  with demos to my repo to hopefully make the direction I am currently intending for the plugin a bit more clear. 

https://github.com/mmnoo/leaflet-declarative-layers
